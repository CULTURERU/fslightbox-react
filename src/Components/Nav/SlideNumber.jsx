import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { ifOnUpdateExistsHandleItForState } from "../../Helpers/State/ifOnUpdateExistsHandleItForState";

const SlideNumber = (
    {
        fsLightbox: {
            data: { totalSlides },
            componentsStates: { slide: slideState },
        }
    }
) => {
    const [slide, setSlide] = useState(1);
    slideState.get = () => slide;
    slideState.set = setSlide;

    useEffect(() => {
        ifOnUpdateExistsHandleItForState(slideState);
    });

    if (totalSlides === 1) {
        return null;
    }

    return (
        <div className="fslightbox-slide-number-container fslightbox-flex-centered">
            <div>{ slide }</div>
            <div className="fslightbox-slash">/</div>
            <div>{ totalSlides }</div>
        </div>
    );
};

SlideNumber.propTypes = {
    fsLightbox: PropTypes.object.isRequired
};
export default SlideNumber;