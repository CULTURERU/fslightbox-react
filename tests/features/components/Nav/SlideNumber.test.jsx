import React from 'react';
import { mount, shallow } from "enzyme";
import { act } from 'react-dom/test-utils';
import SlideNumber from "../../../../src/Components/Nav/SlideNumber";

const totalSlides = 4;
const slideState = {};
const fsLightbox = {
    data: {
        totalSlides: totalSlides
    },
    componentsStates: {
        slide: slideState
    }
};

let slideNumber;

describe('slide - Components state', () => {
    beforeAll(() => {
        slideNumber = mount(<SlideNumber fsLightbox={ fsLightbox }/>);
    });

    it('should be initially 1', () => {
        expect(slideState.get()).toBe(1);
    });

    it('should set to 2', () => {
        act(() => {
            slideState.set(2);
        });
        expect(slideState.get()).toBe(2);
    });

    describe('onUpdate', () => {
        let actionToBeCalled = jest.fn();

        beforeAll(() => {
            slideState.onUpdate = () => {
                actionToBeCalled();
            };
            // updating state
            act(() => {
                slideState.set(3);
            });
        });

        it('should call onUpdate', () => {
            expect(actionToBeCalled).toBeCalled();
        });
    });
});

describe('SlideNumber DOM', () => {
    describe('totalSlides > 1', () => {
        beforeAll(() => {
            fsLightbox.data.totalSlides = 5;
            slideNumber = shallow(<SlideNumber fsLightbox={ fsLightbox }/>);
        });

        describe('slide (first child of wrapper)', () => {
            it('should be initially 1', () => {
                expect(slideNumber.childAt(0).text()).toBe("1");
            });

            it('should be 3, because we set slide state to 3', () => {
                fsLightbox.componentsStates.slide.set(3);
                expect(slideNumber.childAt(0).text()).toBe("3");
            });
        });

        describe('static DOM', () => {
            it('should match snapshot', () => {
                expect(slideNumber).toMatchSnapshot();
            });
        });
    });

    describe('totalSlides === 1', () => {
        beforeAll(() => {
            fsLightbox.data.totalSlides = 1;
            slideNumber = shallow(<SlideNumber fsLightbox={ fsLightbox }/>);
        });

        it('should be null', () => {
            expect(slideNumber.html()).toBeNull();
        });
    });
});