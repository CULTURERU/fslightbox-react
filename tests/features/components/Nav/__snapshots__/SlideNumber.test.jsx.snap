// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideNumber DOM totalSlides > 1 static DOM should match snapshot 1`] = `
<div
  className="fslightbox-slide-number-container fslightbox-flex-centered"
>
  <div>
    3
  </div>
  <div
    className="fslightbox-slash"
  >
    /
  </div>
  <div>
    5
  </div>
</div>
`;
